File: vite-template-greenloop/src/types/donation.ts
export interface DonationRequestDto {
  productId: number;
  description?: string;
  donationLocation?: string;
  donorNote?: string;
}

export interface DonationUpdateDto {
  status?: DonationStatus;
  receiverNote?: string;
  donationLocation?: string;
}

export interface DonationResponseDto {
  id: number;
  title: string;
  description?: string;
  imageUrl?: string;
  donorId: number;
  donorName: string;
  receiverId?: number;
  receiverName?: string;
  productId: number;
  productName: string;
  donationDate: string;
  receivedDate?: string;
  status: DonationStatus;
  donationLocation?: string;
  receiverNote?: string;
  donorNote?: string;
  pointsAwarded?: number;
}

export interface DonationSummaryDto {
  id: number;
  title: string;
  description?: string;
  imageUrl?: string;
  donorName: string;
  productName: string;
  donationDate: string;
  status: DonationStatus;
  donationLocation?: string;
  pointsAwarded?: number;
}

export enum DonationStatus {
  PENDING = "PENDING",
  CONFIRMED = "CONFIRMED",
  IN_PROGRESS = "IN_PROGRESS",
  COMPLETED = "COMPLETED",
  CANCELLED = "CANCELLED",
}

export interface CreateDonationRequest {
  productId: number;
  description?: string;
  donationLocation?: string;
  donorNote?: string;
}

export interface DonationStatistics {
  totalDonationsGiven: number;
  totalDonationsReceived: number;
  totalPointsEarned: number;
  activeDonations: number;
  completedDonations: number;
  pendingRequests: number;
}

File: src/main/java/com/greenloop/greenloop/donation/domain/Donation.java
package com.greenloop.greenloop.donation.domain;

import com.greenloop.greenloop.User.domain.User;
import com.greenloop.greenloop.product.domain.Product;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Donation {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private String title;

    @Column
    private String description;

    @Column
    private String imageUrl;

    @ManyToOne
    @JoinColumn(name = "donor_id")
    private User donor;

    @ManyToOne
    @JoinColumn(name = "receiver_id")
    private User receiver;

    @OneToOne
    @JoinColumn(name = "product_id")
    private Product product;

    @Column(nullable = false)
    private LocalDateTime donationDate;

    @Column
    private LocalDateTime receivedDate;

    @Enumerated(EnumType.STRING)
    private DonationStatus status;

    @Column
    private String donationLocation;

    @Column
    private String receiverNote;

    @Column
    private String donorNote;

    @Column
    private Integer pointsAwarded;

    @PrePersist
    protected void onCreate() {
        donationDate = LocalDateTime.now();
        if (status == null) {
            status = DonationStatus.PENDING;
        }
    }
}